### ISSUE TITLE
Adding support for posting test coverage 

### ISSUE THREAD

**Issue by theScud** [yumemi-inc/danger-swift-kantoku#4]
Hey @el-hoshino  are you already woking on this or can i take a crack at this ? 

Just to make sure we are on the same page , these are the Feature Requirements based on what i have seen in https://github.com/f-meloni/danger-swift-coverage

1. Use xccov via XCResult to read code coverage data from xcresult bundle
2. Give option for user to warn or fail build if coverage isn't found 
3. add option filter out (ignore) code coverage for specific targets ie: test or pods coverage
4. add option to filter and post coverage for changed files. - TBD

**Comment by el-hoshino** [comment]
@theScud Thanks for your investigation!
I just released [0.1.1](https://github.com/yumemi-inc/danger-swift-kantoku/releases/tag/v0.1.1) and basic test coverage feature should be included (actually I implemented it in March and I forgot to release it till now...), so it should be able to parse coverage from xcresult bundle now.

The missing features are
- 2) give option for user to warn or fail build if coverage isn't found ([currently only warning](https://github.com/yumemi-inc/danger-swift-kantoku/blob/8d3283ec9a00f524135e6ccc80762457c1891389/Sources/DangerSwiftKantoku/Kantoku.swift#L125))
- 3) add option filter out (ignore) code coverage for specific targets ie: test or pods coverage, and
- 4) add option to filter and post coverage for changed files ([there is a TODO placeholder though](https://github.com/yumemi-inc/danger-swift-kantoku/blob/7322cb8cde92587fc82859a70594ece5067db07c/Sources/DangerSwiftKantoku/Kantoku%2BCoverage.swift#L53))

I'm busy with my company's work now so if you can implement these for me it would be very appreciated!

Basically all options should be able to found in [XCResultParsingConfiguration](https://github.com/yumemi-inc/danger-swift-kantoku/blob/main/Sources/DangerSwiftKantoku/XCResultParsingConfiguration.swift)

**Comment by theScud** [comment]
cool i'll push something in a week or so

**Comment by theScud** [comment]
are you sure we want to put everything inside XCResultParsingConfiguration , will get kinda bloated no ?

**Comment by el-hoshino** [comment]
@theScud As long as all properties can be set in the public initializer, and there is a default value for each property in that public initializer, it should be fine.

**Comment by theScud** [comment]
@el-hoshino  i have raised a draft PR 
https://github.com/yumemi-inc/danger-swift-kantoku/pull/5/ 

i'll add a few more details in the description.

but i am a little confused about the commit log. i pull your merged main into mine  and cut a new feature branch. Not sure why i had merge conflicts.

**Comment by el-hoshino** [comment]
@theScud If you mean "Merging is blocked" text, don't worry it's not conflict, I just changed the merging settings that it now requires at least 1 approval in code review before merging.

**Comment by theScud** [comment]
@el-hoshino  not exactly when i raised the PR , initially there were merge conflicts , which was strange. My default setting for Git pull is rebase. so could be  just that
